src/library/searchquery.cpp

```diff
@@ -203,10 +200,8 @@ void CrateFilterNode::getTrackIds() {
 
 bool CrateFilterNode::match(const TrackPointer& pTrack) const {
     QString id = pTrack->getId().toString();
-    for (const QString& it : m_trackIds) {
-        if (it == id) {
-            return true;
-        }
+    if (m_trackIds.contains(id)) {
```

Comment:
This is still doing string comparison, it's just not explicit anymore. Make `m_trackIds` a `QList<TrackId>`